config/controller.yaml

```diff
@@ -64,6 +64,8 @@ spec:
           value: config-observability
         - name: METRICS_DOMAIN
           value: knative.dev/serving
+        securityContext:
+          allowPrivilegeEscalation: false
```

Comment:
Can you elaborate on why this is needed to be set explicitly?

Reply:
This stops the user from being able to `sudo` or otherwise elevate privileges, without it starting the container as a non-root user does not mean the user cannot become root afterwards.

A good explanation is here: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privilege-escalation